CMakeLists.txt

```diff
@@ -11,12 +11,15 @@ add_subdirectory(third_party/pybind11)
 set(TENSEAL_BASEDIR tenseal)
 set(SOURCES
     ${TENSEAL_BASEDIR}/binding.cpp
+    ${TENSEAL_BASEDIR}/sealapi.cpp
     ${TENSEAL_BASEDIR}/tensealcontext.cpp
     ${TENSEAL_BASEDIR}/utils.cpp
     ${TENSEAL_BASEDIR}/tensors/bfvnaivevector.cpp
     ${TENSEAL_BASEDIR}/tensors/bfvvector.cpp
     ${TENSEAL_BASEDIR}/tensors/ckksvector.cpp
 )
 
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wnarrowing -Wextra -Wno-varargs -Werror -Wno-strict-overflow -Wfatal-errors -fno-strict-overflow -Wno-psabi -Wno-noexcept-type -Wno-maybe-uninitialized -fstack-protector")
```

Comment:
Wouldn't this stop the build if those flags aren't available on the target platform? I would maybe suggest adding them if available